lib/user-methods.js

```diff
@@ -90,14 +90,17 @@ function fetchWithTimeout(input, init) {
                 url: xhr.responseURL,
                 headers: xhr.responseHeaders
             };
+            if (!options.headers) {
+                options.headers = {'content-type': xhr.getResponseHeader('content-type')};
+            }
             const body = 'response' in xhr ? xhr.response : xhr.responseText;
             resolve(new Response(body, options));
         };
         xhr.onerror = () => reject(new TypeError('Network request failed'));
```

Comment:
It's not directly related to this PR, but it would be nice to flow the original error as we're going to print that if the promise is rejected and it'll be more informative than the vague `Network request failed`. Also, I'm not a js dev, so maybe I'm wrong, but throwing a TypeError feels a bit odd.

Reply:
I have no idea why, but TypeError is actually what the spec says should be thrown.

Probably some historical thing where some library that originally implemented fetch() threw it for anything that didn't result in valid json and that got enshrined at the correct behavior.